app/storyedit/models/story.model.ts

```diff
@@ -69,57 +125,59 @@ export class StoryModel {
     return this.doc.destroy().map(() => { return true; });
   }
 
-  get isValid(): boolean {
-    for (let key in this.invalid) {
-      if (this.invalid[key]) { return false; }
-    }
-    return true;
-  }
-
-  get isChanged(): boolean {
-    for (let key in this.changed) {
-      if (this.changed[key]) { return true; }
+  invalid(fieldName?: string): string {
+    if (fieldName === 'title') {
+      if (!this.title || this.title.length < 1) {
+        return 'is required';
+      } else if (this.title.length < 10) {
+        return 'is too short';
+      }
+    } else if (fieldName === 'shortDescription') {
+      if (!this.shortDescription || this.shortDescription.length < 1) {
+        return 'is required';
+      } else if (this.shortDescription.length < 10) {
+        return 'is too short';
+      }
+    } else if (fieldName === 'genre') {
+      if (!this.genre || this.genre.length < 1) {
+        return 'is required';
+      }
+    } else if (fieldName === 'subGenre') {
+      if (!this.subGenre || this.subGenre.length < 1) {
+        return 'is required';
+      }
+    } else if (fieldName === 'extraTags') {
+      let tags = (this.extraTags || '').replace(/\s*,\s*/, ',').split(',');
+      for (let tag of tags) {
+        if (tag && tag.length < 3) {
+          return `must contain at least 3 characters per tag (see ${tag})`;
+        }
+      }
+    } else if (fieldName === 'tags') {
+      if (this.invalid('genre')) {
+        return `Genre ${this.invalid('genre')}`;
+      } else if (this.invalid('subGenre')) {
+        return `SubGenre ${this.invalid('subGenre')}`;
+      } else if (this.invalid('extraTags')) {
+        return `Tags ${this.invalid('extraTags')}`;
+      }
+    } else if (!fieldName) {
+      return this.invalid('title') || this.invalid('shortDescription') ||
+             this.invalid('genre') || this.invalid('subGenre');
+    } else {
+      return null;
     }
```

Comment:
I started out with that template-driven approach.  It supports "most" of the validations I'd need out-of-the-box.  My hangup was that with the state living in the form, it (1) didn't persist across the 3 subpages, and (2) required a bunch of craziness to pass the valid/changed state up to the parent components "Save/Publish" buttons.

There was a middle-ground option to use the `FormBuilder` api, but I found the syntax a bit verbose.  Encapsulating the state in the model ended up being the cleanest way to share bindings on it.

That said, any simpler form would probably be easier to do with the built-in stuff.  Also, this could probably use (another) refactor to get the validations out of the model, into their own `story.validations.ts` module.